### ISSUE TITLE
using animation.save with ffmpeg writer after compiling with py2exe opens command prompt

### ISSUE THREAD

**Issue by kjgregory** [matplotlib/matplotlib#4021]
I'm using animation.save with ffmpeg as my writer in a wxPython widget and it works great when I run it with python. However, when I compile it into an executable for windows using py2exe, it still works but it opens a command prompt everytime.  It still encodes my video correctly and everything works fine, but I'd like to suppress the command prompt. I'm not really sure why this is happening, so not sure how to suppress it. Here is a snippet of my code:

```python
    Writer = animation.writers['ffmpeg']
    writer = Writer(fps=40,metadata=metadata)
    ims = []
        .... #loop ~120 times generating images using imshow
        ims.append([im1, im2, axtxt])
    anim = animation.ArtistAnimation(self.fig, ims, interval=33)
    anim.save(fname, writer=writer, extra_args=['-c:v', 'libx264', '-preset', 'ultrafast', '-tune', 'stillimage'])
```

FWIW, I had a similar problem with running ffplay using subprocess.Popen() and I solved it by setting 'creationflags=subprocess.CREATE_NO_WINDOW' (0x08000000).

**Comment by tacaswell** [comment]
@jbmohler Any ideas?

**Comment by tacaswell** [comment]
https://stackoverflow.com/questions/24130623/using-python-subprocess-popen-cant-prevent-exe-stopped-working-prompt and links there in.

Basically windows is broken.

**Comment by dopplershift** [comment]
I wonder what happens if we add that `creationflags` argument to our subprocess call... Unfortunately, no windows machines here.

**Comment by kjgregory** [comment]
I'll be happy to test it

**Comment by dopplershift** [comment]
Are you able to just modify the source for matplotlib to test? If so, just change lib/matplotlib/animation.py line 190-192 from:
```
self._proc = subprocess.Popen(command, shell=False,                                        
                               stdout=output, stderr=output,                                
                               stdin=subprocess.PIPE)                                       
```
to
```
self._proc = subprocess.Popen(command, shell=False,                                        
                              stdout=output, stderr=output,                                
                              stdin=subprocess.PIPE, creationflags=0x08000000)                                       
```
Or would it be easier for me to just make a PR--could you test that?

**Comment by kjgregory** [comment]
I'm pretty sure I can edit it. I tried to do so very briefly, but didn't know where to look. I don't have git setup for matplotlib, so I think it will be easier for me to just make the change manually and report back.

**Comment by kjgregory** [comment]
that seems to have fixed it, @dopplershift thanks!

**Comment by dopplershift** [comment]
Alright, then I'll open up a PR to do so in our code.

**Comment by dopplershift** [comment]
Alright, PR is up for anyone who wants to give it a test.

@kjgregory can you confirm that this mimics what worked for you?

**Comment by dopplershift** [comment]
Should be closed by #4041. Re-open if that's not the case.
